xbmc/addons/Skin.cpp

```diff
@@ -500,40 +500,78 @@ void CSkinInfo::SettingOptionsSkinFontsFiller(const SettingConstPtr& setting,
   bool currentValueSet = false;
   std::string strPath = g_SkinInfo->GetSkinPath("Font.xml");
 
+  // Look for fontsets that are defined in the skin's Font.xml file
   CXBMCTinyXML xmlDoc;
-  if (!xmlDoc.LoadFile(strPath))
+  const TiXmlElement* pRootElement;
+  if (xmlDoc.LoadFile(strPath))
   {
-    CLog::Log(LOGERROR, "FillInSkinFonts: Couldn't load {}", strPath);
-    return;
-  }
 
-  const TiXmlElement* pRootElement = xmlDoc.RootElement();
-  if (!pRootElement || pRootElement->ValueStr() != "fonts")
+    pRootElement = xmlDoc.RootElement();
+    if (!pRootElement || pRootElement->ValueStr() != "fonts")
+    {
+      CLog::Log(LOGERROR, "FillInSkinFonts: file {} doesn't start with <fonts>", strPath);
+    }
+    else
+    {
+      const TiXmlElement* pChild = pRootElement->FirstChildElement("fontset");
+      while (pChild)
+      {
+        const char* idAttr = pChild->Attribute("id");
+        const char* idLocAttr = pChild->Attribute("idloc");
+        if (idAttr != nullptr)
```

Comment:
```suggestion
        if (idAttr)
```

You do the same below, so please try to stick with the same style